### DISCUSSION TITLE
Safest upgrade path

### DISCUSSION THREAD

**Discussion by jsalts** [nats-io/nats.java#651] [General]
Hello, 

We are on jnats 2.6.7 and nats server 2.1.2. We'd like to upgrade to the latest client and server version. Could you clarify what the safest upgrade path for us would be? This is in regards to the notes below (we don't use durable queue subscriptions). 

`Version 2.11.6 is the last java-nats version which is supported to work with server v2.3.4 and earlier. It will not be officially supported to work with servers after v2.3.4, but should be fine if you don't use the queue behavior advertised in example code NatsJsPushSubQueueDurable.java and provided with java-nats 2.11.5.`

**Comment by scottf** [comment]
@jsalts So server 2.1.2 does not support Jetstream so you have no issue about data storage. The server protocol has not changed, and there has only been minor work to the client in regard to core functionality, all of which is either bug fix or improvement. I think there should not be any issue just simply upgrading to the latest of both the client and server.

**Reply by jsalts** [reply]
Thanks!
